dataprofiler/profilers/profile_builder.py

```diff
@@ -1550,6 +1572,22 @@ def tqdm(l):
         # Newly introduced features (python3.8) improves the situation
         sample_ids = np.array(sample_ids)
 
+        # Record the previous mean/std values of columns that need to
+        # have correlation updated. Non-numeric columns have NaN
+        # so that they stay NaN when updating.
+        prev_dependent_properties = {
+            'mean': np.full(len(self._profile.keys()), np.nan),
+            'std': np.full(len(self._profile.keys()), np.nan)
+        }
+        for id, col in enumerate(self._profile.keys()):
```

Comment:
since this code and code above are very similar, can we refactor such that we don't repeat?